src/main/java/com/example/bookshop/controller/CategoryController.java

```diff
@@ -0,0 +1,76 @@
+package com.example.bookshop.controller;
+
+import com.example.bookshop.dto.BookDto;
+import com.example.bookshop.dto.CategoryDto;
+import com.example.bookshop.mapper.BookMapper;
+import com.example.bookshop.repository.BookRepository;
+import com.example.bookshop.service.CategoryService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Category management", description = "Endpoints for categories")
+@RestController
+@RequestMapping("/categories")
+public class CategoryController {
+    @Autowired
+    private CategoryService categoryService;
+    @Autowired
+    private BookMapper bookMapper;
+    @Autowired
+    private BookRepository bookRepository;
+
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @Operation(summary = "Create category", description = "Create new category")
+    @PostMapping
+    public CategoryDto createCategory(@RequestBody @Valid CategoryDto categoryDto) {
```

Comment:
```suggestion
    public CategoryDto createCategory(@RequestBody @Valid CreateCategoryRequestDto categoryDto) {
```
Let’s  create separate request dto and validate fields

Reply:
Not fixed

Reply:
Don't provide CategoryDto response with id as a request
fix comment